src/logger.c

```diff
@@ -90,17 +96,31 @@ static struct log_file pot = {NULL, NULL, NULL, 0, -1};
 
 static int in_logger = 0;
 
-static void log_file_init(struct log_file *f, char *name, int size)
+static void log_file_init(struct log_file *f, char *name, char *perms, int size)
 {
+	p_u = "0600";
+	p_ug = "0660";
+	p_ugo = "0666";
+	if (!((strcmp(perms, p_u) == 0) || (strcmp(perms, p_ug) == 0) || (strcmp(perms, p_ugo) == 0)))
+		pexit("PotFilePerms or LogFilePerms %s invalid, valid values: %s, %s, %s\n\n", perms, p_u, p_ug, p_ugo);
+	perms_t = strtoul(perms, NULL, 8);
```

Comment:
This if might be better.

int bad = 0;
if (perms_t & 0x49)
   bad = 1;  //  an execute bit is set
if (!(perms_t & 0x92))
   bad = 1;  // no write bit set.
if (bad) {
   // perform bad exit
}

This would make sure that there is no execute bit  (0x49 is 0b01001001  (i.e exe bits), and 0x92 is 0b010010010 (write bits).